projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -1110,8 +1110,9 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
         const colLength = this.columnList.length;
         const topCols = this.columnList.filter((item) => colsArray.indexOf(item) === -1);
         if (topCols.length > 0) {
+            const columnsChanged = this.columns.length !== colLength;
```

Comment:
How about we update `colLength` above to cache the `this.columns.length` instead so the compare afterwards makes sense? Seems to be used just for that and comparing the columnList against the top-level filtered length is an odd check for changes.